### DISCUSSION TITLE
SetOption114 and Button Leds

### DISCUSSION THREAD

**Discussion by erdembey** [arendst/Tasmota#17072] [Q&A] [answered]
Hello,

I am using several tasmota devices in my home.
My latest addition was a sonoff TX T0 3 gang switch.

My intention is to use only 1 button with relay and the others to control other systems over HA.
Right now, when i use a rule which contains SetOption114 based on button state i am loosing status light changes on switch. Also it seems its not sending status changes to HA.

Can any one provide a sample rule for this stiutation ?

**Comment by barbudor** [comment, marked as answer]
Using SetOption114 is for "Switches" while the Tasmota template for Sonoff TX T0 uses "Buttons"
May be you should start by stating exactly where what you've done differs from the defaults

> when i use a rule which contains SetOption114

This doesn't make sense. SetOption114 is a configuration command which has nothing to do in a rule. Once applied, all "Switches" are detached from relays.

What do you want Tasmota to send to HA for the 2 buttons (or switches ?) for remote ?
Depending what you exactly want to achieve, different solutions are possible.
Starting by removing the GPIO Relays for the relays you don't want to use (or replacing by OUTPUT_LOW to be sure they are never activated)

**Reply by erdembey** [reply]
First of all thank you for your reply.

Let me explain my scenario.

I want to use buton1 for open/close our normal room light.

Button2 will be used to trigger on/off function of sonoff basic switch which will be connected to our desk lamps.

Button3 will be used to trigger activation on room heating system.

Both button2 and button3 will / may work over HA.

Btw, it will be great if we can see the status of buttons via lighting of buttons.

**Reply by barbudor** [reply]
I have no idea of how HA is working so if you don't tell me what exactly Tasmota should send to HA, I won't be able to help you.

Now you can have Button 2 to talk directly to the Sonoff if it is running Tasmota like:
```
Rule1 on button3 do publish cmnd/sonoffdesklamp/POWER toggle endon
```

If the heating system if using Tasmota, you can use the same trick.

As the template shows, the status leds are not under the control of Tasmota. They are likely to be driven by the same GPIO as the Relay which is very common in that type of wall switches. So if the relay is not actuated, no way to have the led to change.
The simplest would be to keep the relay working. Nice feedback to the user to hear the relay clicking 🙄

**Reply by erdembey** [reply]
Alright i've used it as it is. Thank you for your feedback, as you've stated in your message keeping the feedback is best option for us.
I am using button 2 and 3 over HA to trigger my desk lamp and heating system.
